packages/components/card/src/card.ts

```diff
@@ -0,0 +1,102 @@
+import { property, queryAssignedElements } from 'lit/decorators.js';
+import styles from './card.scss.js';
+
+export type CardHeightOptions = 'fixed' | 'flex';
+export type CardOrientation = 'horizontal' | 'vertical';
+export type CardMediaPosition = 'start' | 'end';
+
+/**
+ * Use cards to display media and text in a compact, appealing way.
+ *
+ * ```html
+ * <sl-card></sl-card>
+ * ```
+ *
+ * @cssprop --sl-card-media-aspect-ratio - The aspectratio of the media container (default is 4/3). By default this ratio is always maintained, and will cause the media to become smaller when there isn't sufficient space for the full width.
+ * @cssprop --sl-card-media-width - The width of the media in relation to the text. Can be set in pixels or `fr`.
+ * @cssprop --sl-card-media-x - X-Focuspoint of the media; this is taken as the center when the media is cropped.
+ * @cssprop --sl-card-media-y - Y-Focuspoint of the media; this is taken as the center when the media is cropped.
+ * @cssprop --sl-card-orientation-breakpoint - When card is smaller than this size it will switch from horizontal (when set) to vertical layout.
+ * @cssprop --sl-card-stretch-image - Set this to 100% when the aspectratio of the media doesn't matter and you want it to fill the full height of the card.
+ * @cssprop --sl-card-text-width - The width of the text in relation to the media. Can be set in pixels (not recommended) or `fr`.
+ *
+ * @slot default - Title of the card
+ * @slot media - Media, this can be an image or video
+ * @slot header - Subtitle or badges
+ * @slot body - Body text of the card
+ * @slot actions - Icon button for actions on the card.
+ */
+export class Card extends LitElement {
+  /** @private */
+  static override styles: CSSResultGroup = styles;
+
+  /** Observe the grid width. */
+  #resizeObserver?: ResizeObserver = new ResizeObserver(() => {
+    this.#setOrientation();
+  });
+
+  /** @private The slotted media. */
+  @queryAssignedElements({ slot: 'media' }) media?: HTMLElement[];
+
+  @property({ type: Boolean, reflect: true }) padding: boolean = false;
+  @property({ reflect: true }) height: CardHeightOptions = 'fixed';
+  @property({ reflect: true }) orientation: CardOrientation = 'horizontal';
+  @property({ reflect: true }) icon?: string;
+  @property({ reflect: true, attribute: 'media-position' }) mediaPosition: CardMediaPosition = 'start';
+
+  override connectedCallback(): void {
+    super.connectedCallback();
+    this.#setOrientation();
+
+    this.#resizeObserver?.observe(this);
+  }
+
+  override disconnectedCallback(): void {
+    this.#resizeObserver?.disconnect();
+    this.#resizeObserver = undefined;
+
+    super.disconnectedCallback();
+  }
+
+  override updated(changes: PropertyValues<this>): void {
+    super.updated(changes);
+
+    if (changes.has('orientation')) {
+      this.#setOrientation();
+    }
+  }
+
+  override render(): TemplateResult {
+    return html`
+      <div class="container">
```

Comment:
Why have a container div when you also have `:host`?

Reply:
I think i added it when i wanted to use container queries, but that didn't end up working.

Reply:
You may have run into a bug/regression in Safari? That was fixed recently. It definitely should work just fine on `:host`.

Reply:
no, container queries only work when the container has an explicit size, the :host couldn't have that, so my solution was to use a container. But with all the different combinations of overflow and flexible height i couldn't get it to work with container queries. Our requirements were the issue, not the support of container queries.